tests/ops/op_math/test_controlled_decompositions.py

```diff
@@ -133,6 +133,41 @@ def expected_circuit():
 
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("control_wires", ([1], [1, 2], [1, 2, 3]))
+    def test_decomposition_circuit_gradient(self, control_wires, tol):
+        """Tests that the controlled decomposition of a single-qubit operation
+        behaves as expected in a quantum circuit"""
+        n_qubits = 4
+        np.random.seed(1337)
+
+        dev = qml.device("default.qubit", wires=n_qubits)
+        init_state = np.random.rand(2**n_qubits) + 1.0j * np.random.rand(2**n_qubits)
+        init_state /= np.sqrt(np.dot(np.conj(init_state), init_state))
+        init_state = np.array(init_state)
+        target_wires = [0]
+        control_values = [True] * len(control_wires)
+
+        def circuit(p):
+            qml.StatePrep(init_state, wires=range(n_qubits))
+            qml.ctrl(
+                qml.Rot(*p, wires=target_wires),
+                control_wires,
+                control_values=control_values,
+            )
+            return qml.probs(wires=target_wires)
+
+        circ_ad = qml.QNode(circuit, dev, diff_method="adjoint")
+        circ_ps = qml.QNode(circuit, dev, diff_method="finite-diff")
```

Comment:
finite diff?

Reply:
Oups. This originates from Lightning and it was meant to be parameter-shift at first but 
```
pennylane.operation.OperatorPropertyUndefined: The operation C(Rot) does not have a grad_recipe, parameter_frequencies or a generator defined. No parameter shift rule can be applied.
```
But we can use `backprop` here since this is `default.qubit`. Let me change it.